src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Compilation.cs

```diff
@@ -419,6 +419,10 @@ public GenericDictionaryLookup ComputeGenericLookup(MethodDesc contextMethod, Re
             return GenericDictionaryLookup.CreateHelperLookup(contextSource, lookupKind, targetOfLookup);
         }
 
+        // CoreCLR compat - referring to function pointer types handled as IntPtr. No MethodTable for function pointers for now.
+        private static TypeDesc WithoutFunctionPointerType(TypeDesc type)
```

Comment:
Are there more places where this should be done for consistency?

For example, does enumeration of method argument types using reflection return IntPtr too?

Reply:
We don't have tests so I would rather not spend time on that. We also don't have test for arrays of function pointers that do something atrocious. This is just to make a test pass.

We could also just delete it because the test is very likely not testing what its author intended, or block it on #11354.